README.md

```diff
@@ -110,89 +152,65 @@ execute the following in any OCaml top-level:
-used to evaluate our `byteweight` algorithm for finding symbols inside
-the binary. It is also a supporting toolkit for byteweight
-infrastructure, it can download, create and install binary signatures,
-used for identification.
+To get information about specific plugin named `<PLUGIN>` use
+`--<PLUGIN>-help` option, e.g., `bap --llvm-help`.
 
+The `bap` utility works with the whole binaries, if you have just few
+bytes with which you would like to tackle, then `bap-mc` is what you
+are looking for.
 
-## Using from other languages
+## RPC
 
-BAP exposes most of its functionality using `JSON`-based RPC protocol,
+Some of BAP functionality is exposed via `JSON`-based RPC protocol,
 specified
 [Public API Draft](https://github.com/BinaryAnalysisPlatform/bap/wiki/Public-API-%5Bdraft%5D)
-doument. The protocol is implemented by `bap-server` program that is
-shipped with bap by default. You can talk with server using `HTTP`
-protocol, or extend it with any other transporting protocol you would
-like.
+document. The protocol is implemented by `bap-server` program, that
+can be installed with `opam install bap-server` command. You can talk
+with server using `HTTP` protocol, or extend it with any other
+transporting protocol you would like.
 
-## Extending BAP
 
-We're always welcome for any contributions. If you want to add new
-code, or fix a bug, feel free to clone us, and create a pull request.
+# Learning
 
-But BAP can also be extended in a non invasive way, using plugin
-system. That means, that you can use `bap` library, to extend the
-`bap` library! See our
-[blog](http://binaryanalysisplatform.github.io/bap_plugins/) for more
-information.
-
-## Learning BAP
-
-Other than [API](http://binaryanalysisplatform.github.io/bap/api/v0.9.7/Bap.Std.html) documentation, we have [blog](http://binaryanalysisplatform.github.io/bap_plugins/) and
+Other than
+[API](http://binaryanalysisplatform.github.io/bap/api/v1.0.0/argot_index.html)
+documentation, we have
+[blog](http://binaryanalysisplatform.github.io/bap_plugins/) and
 [wiki](https://github.com/BinaryAnalysisPlatform/bap/wiki/), where you
 can find some useful information. Also, we have a permanently manned
-chat in case of emergency. Look at the badge on top of the README file,
-and feel free to join.
+chat in case of emergency. Look at the badge on top of the README
+file, and feel free to join.
+
+# Contributing
 
-# License
+BAP is a framework, so you don't need to change its code to extend
+it. We use dependency injection principle, with lots of injection
+points, that allow you to alter BAP behavior. However, bugs happen, so
+if you any problems, questions or suggestions, please, don't hesitate
+to use our issue tracker. Submitting a pull request with a problem fix
+will make us really happy.
 
-Please see the `LICENSE` file for licensing information.
+If you wrote analysis with BAP, then don't hesitate to release it to
+opam, for the benefit of the community.
```

Comment:
If we're not releasing to the main `opam` repository anymore, perhaps we should use our `opam-repository` as the place where users submit pull requests to have their plugins listed rather than the main one? If so, we should explain how to do this explicitly.